src/analysis/analytic/basic.lean

```diff
@@ -0,0 +1,391 @@
+/-
+Copyright (c) 2020 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import analysis.calculus.times_cont_diff tactic.omega analysis.complex.exponential
+analysis.specific_limits
+
+/-!
+# Analytic functions
+
+A function is analytic in one dimension around `0` if it can be written as a converging power series
+`Σ pₙ zⁿ`. This definition can be extended to any dimension (even in infinite dimension) by
+requiring that `pₙ` is a continuous `n`-multilinear map. In general, `pₙ` is not unique (in two
+dimensions, taking `p₂ (x, y) (x', y') = x y'` or `y x'` gives the same map when applied to a
+vector `(x, y) (x, y)`). A way to guarantee uniqueness is to take a symmetric `pₙ`, but this is not
+always possible in nonzero characteristic (in characteristic 2, the previous example has no
+symmetric representative). Therefore, we do not insist on symmetry or uniqueness in the definition,
+and we only require the existence of a converging series.
+
+The general framework is important to say that the exponential map on bounded operators on a Banach
+space is analytic, as well as the inverse on invertible operators.
+
+## Main definitions
+
+Let `p` be a formal multilinear series from `E` to `F`, i.e., `p n` is a multilinear map on `E^n`
+for `n : ℕ`.
+
+* `p.radius`: the largest `r : ennreal` such that `∥p n∥ * r^n` grows subexponentially, defined as
```

Comment:
In higher dimension we need more information than just one number to describe the set where the series converges. AFAIR, we have a radius in each direction, and they're convex in log-scale. If it's hard to formalize now, I think it's worth mentioning in the docstring (I usually create a section called TODO).

Reply:
I have mentioned this in the file docstring.